### ISSUE TITLE
incompatibility with newer gnutls

### ISSUE THREAD

**Issue by ariep** [GothenburgBitFactory/taskwarrior#2138]
After upgrading my system gnutls, `task sync` no longer works (see below).

 #### To report a bug...

* What command(s) did you run?

`task sync`

* What did you expect to happen?

Succesful sync with my taskd server.

* What actually happened?

```
Syncing with REDACTED:53589

Handshake failed.  Error in the certificate.
Sync failed.  Could not connect to the Taskserver.
```

* Paste the output of the ```task diag``` command.

```
task 2.5.1
   Platform: Linux

Compiler
    Version: 8.2.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Apr  7 2019 15:13:14
      CMake: 3.9.6
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.6.7
 Build type: Gentoo

Configuration
       File: /home/arie/.taskrc (found), 1560 bytes, mode 100644
       Data: /home/arie/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: /usr/bin/nvim
     Server: REDACTED:53589
         CA: /home/arie/.task/ca.cert.pem, readable, 2411 bytes
      Trust: strict
Certificate: /home/arie/.task/arie.cert.pem, readable, 2452 bytes
        Key: /home/arie/.task/arie.key.pem, readable, 3664 bytes
    Ciphers: NORMAL
      Creds: REDACTED / arie /*************************************

Hooks
     System: Enabled
   Location: /home/arie/.task/hooks
             (-none-)

Tests
      $TERM: screen-256color (120x30)
       Dups: Scanned 305 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 305 tasks for broken references:
             No broken references found
```

This is with gnutls version 3.6.7 ([latest stable in Gentoo](https://packages.gentoo.org/packages/net-libs/gnutls)), and 3.6.6. Downgrading to gnutls 3.5.19 makes the problem disappear again.

To be honest I don't know if this is a bug in gnutls, or if taskwarrior fails to deal with intended but changed behaviour in gnutls.

**Comment by pbeckingham** [comment]
Thank you.

When you upgraded, was the previous GnuTLS removed?  Did you recompile Taskwarrior/Taskserver against the new GnuTLS?

**Comment by pbeckingham** [comment]
I cannot reproduce this with GnuTLS 3.6.7, and recompiling both Taskserver and Taskwarrior.

**Comment by pbeckingham** [comment]
Closing to unresponsive submitter.

**Comment by Tranquility** [comment]
@pbeckingham I also have this problem. I am also using gentoo and I recompiled both the client and the server after updating gnutls to 3.6.7.

**Comment by ariep** [comment]
Recently I had the opportunity to explore this again, when moving my taskd server to another machine, and I think I may know what went wrong here.

I tried to set up a new taskd server with my old certificates from 2015. I ran into the same errors when syncing, both with newer clients and with old ones (that I didn't upgrade because of the above issue). While debugging with `gnutls-cli` both on client and server, I got an error that indicated that some aspect of the certificate was regarded as too old and unsafe by the new taskd server.

So I generated new certificates, and now my new taskd server happily serves clients both with newer and older versions of gnutls (although all clients of course need the new certificates). I suspect that the new certificates would also have solved the original issue, but I'm not really motivated to resurrect my old server to try this.
